lib/index.js

```diff
@@ -373,6 +374,18 @@ class Generator extends EventEmitter {
     this._running = true;
     this.emit('run');
 
+    const restArgs = Object.keys(_.clone(this)).filter(orginalArg => {
+      return !this.orginalArgs[orginalArg] && this[orginalArg] &&
+        (orginalArg.substring(0, 1) !== '_') && (orginalArg !== 'orginalArgs');
+    });
+
+    this.env.getArgument = function (name) {
```

Comment:
This line here is problematic as this [environment object](https://github.com/yeoman/environment) is shared between all generator instances. With this code, only the last ran generator will be available through `getArguments`.

Maybe a way to access a generator instance would be more flexible?